bundler/spec/realworld/double_check_spec.rb

```diff
@@ -37,4 +37,12 @@
     ruby cmd
     ruby cmd
   end
+
+  it "deals with old marshaled specs with nil required_rubygems_version" do
+    install_gemfile <<~G
+      source "https://rubygems.org"
+
+      gem "ruport", "=1.7.0.3", :source => "https://rubygems.manageiq.org"
```

Comment:
I'm not sure this is a good idea in specs, only because there's a potential it could go away in the future (unless this doesn't actually hit the live system? I see some recorded http things below).  For right now though, it's probably fine.

For testing purposes does rubygems have a third party package registry. If not, maybe Github package registry could be used for that purpose and dedicated to testing rubygems/bundler?

Reply:
@Fryguy Correct, this doesn't hit the live system, it uses the recorded responses. Normally I tried to avoid realworld specs like this one, I only use them when I cannot find another way to test what I want to test. In this case, the tricky part was that our current spec setup does not support setting up multiple gem servers when some of them support new APIs and some of them don't. But I will give it another try.

Regarding a testing package registry,  I guess it could make sense although it feels slightly overkill for now given how little use we make of this realworld test harness.

Reply:
That's fair. As the owner of that gem server, we have no intention of removing anything, since we use that for production builds, so for now it's fine, but who knows what could happen in the future.

Thanks for the quick fix!

Reply:
I gave it another try and I managed to write a better spec, it should be ok now!